include/zephyr/bluetooth/classic/hfp_hf.h

```diff
@@ -168,16 +427,492 @@ struct bt_hfp_hf_cb {
  */
 int bt_hfp_hf_register(struct bt_hfp_hf_cb *cb);
 
-/** @brief Handsfree client Send AT
+/** @brief Create the hfp hf session
+ *
+ *  Create the hfp hf session
+ *
+ *  @param conn ACL connection object.
+ *  @param hf Created HFP HF object.
```

Comment:
This is missing details on the validity of the allocated object. Is it possible that it becomes invalid while the app holds onto the pointer? What actions (e.g. a disconnection) may render it invalid? Should it perhaps be reference counted, similar to e.g. `bt_conn` objects?

This is also missing information if this is a synchronous call or an asynchronous one. I.e. do you have a real connection when this function returns, or do you need to wait for some callback?

Reply:
Updated